test/test.js

```diff
@@ -72,54 +76,67 @@ describe('Filter', function() {
 
     afterEach(co.wrap(function* () {
       yield input.dispose();
+      if (output) {
+        yield output.dispose();
+      }
     }));
 
     it('throws if base Filter class is new-ed', function() {
-      expect(() => new Filter('.')).to.throw(TypeError, /abstract class and must be sub-classed/);
+      expect(() => new Filter(input.path())).to.throw(TypeError, /abstract class and must be sub-classed/);
     });
 
     it('throws if `processString` is not implemented', function() {
-      expect(() => new IncompleteFilter('.').processString('foo', 'fake_path')).to.throw(Error, /must implement/);
+      expect(() => new IncompleteFilter(input.path()).processString('foo', 'fake_path')).to.throw(Error, /must implement/);
     });
 
-    it('processes files with extensions included in `extensions` list by default', function() {
-     let filter = new MyFilter('.', { extensions: ['c', 'cc', 'js']});
+    it('processes files with extensions included in `extensions` list by default', co.wrap(function* () {
+     let filter = new MyFilter(input.path(), { extensions: ['c', 'cc', 'js']});
+
+     output = createBuilder(filter);
+     yield output.build();
 
       expect(filter.canProcessFile('foo.c')).to.equal(true);
       expect(filter.canProcessFile('test.js')).to.equal(true);
       expect(filter.canProcessFile('blob.cc')).to.equal(true);
       expect(filter.canProcessFile('twerp.rs')).to.equal(false);
-    });
+    }));
 
-    it('getDestFilePath returns null for directories when extensions is null', function() {
+    it('getDestFilePath returns null for directories when extensions is null', co.wrap(function* () {
       let inputPath = input.path();
       let filter = new MyFilter(inputPath, { extensions: null });
-      filter.inputPaths = [inputPath];
+
+      output = createBuilder(filter);
+      yield output.build();
 
       expect(filter.getDestFilePath('a/bar')).to.equal(null);
       expect(filter.getDestFilePath('a/bar/bar.js')).to.equal('a/bar/bar.js');
-    });
+    }));
 
-    it('getDestFilePath returns null for directories with matching extensions', function() {
+    it('getDestFilePath returns null for directories with matching extensions', co.wrap(function* () {
```

Comment:
Since we support node >= 8, we don't need to use `co` at all, we can now use `async/await` directly. Not only will this be nice, but it will provide better stack traces and debugging ergonomics, mind updating?

Reply:
Created a PR for it https://github.com/stefanpenner/broccoli-persistent-filter/pull/183